backend/schema/corpora_schema_h5ad_implementation.md

```diff
@@ -0,0 +1,90 @@
+# Corpora Schema AnnData Implementation
+
+Authors: mkinsella@chanzuckerberg.com
+
+Document Status: _Approved_
+
+Version: 0.1.0
+
+Date Last Modified: 2020-07-21
+
+Corpora datasets are stored in the HDF5-backed AnnData format as written by version 0.7 of the anndata library. One
+goal of using this format is to have all metadata stored in the same file as the data, so cellxgene will have access to
+all Corpora metadata within a single file.
+
+The Corpora schema requirements and definitions for the AnnData `X`, `uns`, `obs`, and `obsm` attributes are below.
+
+
+### Note on types
+The types below are python3 types. Note that a python3 `str` is a sequence of Unicode code points, which is stored
+null-terminated and UTF-8-encoded by anndata.
+
+
+## `X`
+
+Corpora does not impose any additional constraints on the `X` data matrix. So it may be sparse or dense and any
+numeric `numpy.dtype`.
+
+Corpora does require that raw counts be present in the h5ad. If the `X` data matrix is not raw counts, then the h5ad
+must have a `raw` attribute where the raw counts are stored in `raw.X`
+
+## `uns`
+
+Recall that `uns` is a mapping with `str`s as keys. Corpora requires the following keys and values in `uns`:
+
+**Key**|**Value Type**|**Notes**
+-----|-----|-----
+version|`dict` with keys "corpora_schema_version" and "corpora_encoding_version" indicating the version of the schema and schema encoding used to create the h5ad.
```

Comment:
what is the value type for the version keys?   Semver string, eg, "X.Y.Z"?

Reply:
also, should each field in this "encoding" spec be defined in the corpora schema spec?  Most are, but not this one.  That may be the right location to define the version type/encoding?

Reply:
👍 Added the SemVer requirement and version to the schema.